src/sql/src/plan/statement/ddl.rs

```diff
@@ -3875,6 +3875,13 @@ fn kafka_sink_builder(
             )?;
             let expr = expr.lower_uncorrelated()?;
 
+            let function_type = expr.typ(&value_desc.typ().column_types);
+
+            if function_type.nullable {
+                scx.catalog
+                    .add_notice(PlanNotice::KafkaSinkNullablePartitionByFunction);
+            }
+
```

Comment:
Determining nullability is difficult, so we can sometimes conclude an expr is nullable, if it is not in reality. I think with the restricted set of queries that are allowed here we should be fine.